middleware/middleware.go

```diff
@@ -0,0 +1,54 @@
+package middleware
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/go-chi/chi"
+	"go.uber.org/zap"
+)
+
+// Middlewarer lets you add middlewares
+type Middlewarer interface {
+	Middleware(middlewares ...func(next http.Handler) http.Handler)
+}
+
+// Flagger lets you create command line flags and functions that use them.
+type Flagger interface {
+	Flag(name string, short string, description string, defaultValue interface{})
+	PreStart(f func())
+}
+
+// DefaultChain sets up the default middlewares conveniently chained together
+// into a single easy-to-add handler.
+func DefaultChain(next http.Handler) http.Handler {
+	// Note: logger goes before recovery so that recovery can use it. We don't
+	// expect the logger to cause panics.
+	return chi.Chain(
+		OpenTracing,
+		Logger,
+		Recovery(func(ctx context.Context, err error, request string) {
+			log := GetLogger(ctx)
+			log = log.With(zap.Error(err))
+			log.With(
+				zap.String("http.request", string(request)),
```

Comment:
`request` is already a string, no?

Reply:
Unfortunately it's `[]byte` so we have to cast it.

Reply:
Isn't it a `string` here https://github.com/danielgtaylor/huma/pull/11/files/f1f9c8507156c28ef756697193edc96fcbb51cea#diff-ce31218ed9b14d108ecc98733ac0abd2R30 ?

Reply:
Oh, duh, it's already cast in the `Recovery` middleware itself. This was a copy/paste error. Will fix.